test/controllers/registrations_controller_test.rb

```diff
@@ -116,4 +116,79 @@ class RegistrationsControllerTest < ActionController::TestCase
 
     assert_redirected_to course_registrations_path(@cs301)
   end
+
+  def attempt_register_user(user, role)
```

Comment:
ISTM you probably want to add two optional parameters here, `attempt_register_user(user, role, course=@cs101, section=@section)`, so that if you need to generalize the uses of this method slightly in your tests, it's easy enough...